chia/util/recursive_replace.py

```diff
@@ -1,12 +1,18 @@
 from __future__ import annotations
 
-from dataclasses import replace
+from dataclasses import is_dataclass, replace
 from typing import Any
 
 
 def recursive_replace(root_obj: Any, replace_str: str, replace_with: Any) -> Any:
     split_str = replace_str.split(".")
     if len(split_str) == 1:
-        return replace(root_obj, **{split_str[0]: replace_with})
+        if is_dataclass(root_obj):
+            return replace(root_obj, **{split_str[0]: replace_with})
+        else:
+            return root_obj.replace(**{split_str[0]: replace_with})
```

Comment:
I'm not sure if it is overall better or not, but it would be more flexible to just use the `.replace()` method if present.  Then even dataclasses can choose to override it.  Might end up useful as a transitional step somewhere?  Though, that hazards accidentally adding one for another purpose and accidentally changing the behavior.  Could decorate them with a marker of some sort...  Maybe some part of this rambling is an interesting though...

Reply:
yeah, I did that in an earlier version of this patch. I don't have a strong opinion either way, I changed it to this after reading up on people on the internet expecting `dataclass` to add a member `replace()` and asking about why it doesn't.

Reply:
I guess folks that want that could add `replace = dataclasses.replace` to the class definition.  I personally am glad that it doesn't take over lots of "regular" methods and sticks to helping out with dunder methods.  Makes it more like a nice way to write a "regular class" than a big ol' framework that takes control of your class interface.  Makes it easier to attach multiple sets of functionality to a single class without them colliding since it mostly exposes the metadata for access as needed.